source/non_matching/coupling.cc

```diff
@@ -123,19 +158,7 @@ namespace NonMatching
             if (ocell->is_locally_owned())
               {
                 ocell->get_dof_indices(odofs);
-                for (unsigned int i=0; i<odofs.size(); ++i)
-                  {
-                    const auto comp_i = space_fe.system_to_component_index(i).first;
-                    if (space_gtl[comp_i] != numbers::invalid_unsigned_int)
-                      {
-                        for (unsigned int j=0; j<dofs.size(); ++j)
-                          {
-                            const auto comp_j = immersed_fe.system_to_component_index(j).first;
-                            if (immersed_gtl[comp_j] == space_gtl[comp_i])
-                              sparsity.add(odofs[i],dofs[j]);
-                          }
-                      }
-                  }
+                constraints.add_entries_local_to_global(odofs, dofs, sparsity);
```

Comment:
I guess you want to add the `dof_mask` here, don't you?

Reply:
of course... 

I'll also add a test with a locally refined mesh, so that this path is tested..

Reply:
It turns out that the version with the dof_mask is not implemented for a standard matrix, but only for a block matrix... I'm not too familiar with that part of the code, so I'll just let this without the dof_mask for the time beeing...